[wasm][hotreload] Cannot build `ApplyUpdateReferencedAssembly` with source link

I can't build `src/mono/wasm/debugger/tests/ApplyUpdateReferencedAssembly/ApplyUpdateReferencedAssembly.csproj` with `/p:ContinuousIntegrationBuild=true`. It fails with:

```
 initializing ChangeMakerService with capabilities: Baseline, AddMethodToExistingType, AddStaticFieldToExistingType, AddInstanceFieldToExistingType, NewTypeDefinition, ChangeCustomAttributes
  baseline ready
  got a change
  parsing patch #1 from /__w/1/s/src/mono/wasm/debugger/tests/ApplyUpdateReferencedAssembly/MethodBody1_v1.cs and creating delta
  Found changes in MethodBody1.cs
EXEC : error : Expected only one module in the delta, got 0 [/__w/1/s/src/mono/wasm/debugger/tests/ApplyUpdateReferencedAssembly/ApplyUpdateReferencedAssembly.csproj]
##[error]EXEC(0,0): error : Expected only one module in the delta, got 0
/__w/1/s/.packages/microsoft.dotnet.hotreload.utils.generator.buildtool/1.0.2-alpha.0.21579.1/build/Microsoft.DotNet.HotReload.Utils.Generator.BuildTool.targets(56,5): error MSB3073: The command ""/__w/1/s/.dotnet/dotnet" /__w/1/s/.packages/microsoft.dotnet.hotreload.utils.generator.buildtool/1.0.2-alpha.0.21579.1/build/../tools/net6.0/Microsoft.DotNet.HotReload.Utils.Generator.BuildTool.dll -msbuild:/__w/1/s/src/mono/wasm/debugger/tests/ApplyUpdateReferencedAssembly/ApplyUpdateReferencedAssembly.csproj -script:/__w/1/s/src/mono/wasm/debugger/tests/ApplyUpdateReferencedAssembly/deltascript.json -p:Configuration=Debug -p:RuntimeIdentifier=browser-wasm" exited with code 10. [/__w/1/s/src/mono/wasm/debugger/tests/ApplyUpdateReferencedAssembly/ApplyUpdateReferencedAssembly.csproj]
```

.. `error : Expected only one module in the delta, got 0`.

This seems to happen specifically when `$(DisableSourceLink)=false`, which gets set because of `$(ContinuousIntegrationBuild)=true`.

The reasons might be similar to https://github.com/dotnet/runtime/issues/62551 .
- I am building this locally, where my branch HEAD is not pushed to any branch

--- a/src/mono/browser/debugger/tests/ApplyUpdateReferencedAssembly/ApplyUpdateReferencedAssembly.csproj
+++ b/src/mono/browser/debugger/tests/ApplyUpdateReferencedAssembly/ApplyUpdateReferencedAssembly.csproj
@@ -14,7 +14,7 @@
     <EnableDefaultCompileItems>false</EnableDefaultCompileItems>
     <EmbedAllSources>true</EmbedAllSources>
 
-    <!-- FIXME: issue -->
+    <!-- workaround to avoid this error: "Expected only one module in the delta, got 0" -->
     <DisableSourceLink>true</DisableSourceLink>
   </PropertyGroup>
 

